lib/diaspora_federation/entity.rb

```diff
@@ -319,8 +319,8 @@ def simple_node(doc, name, value)
       when :integer
         text.to_i if text =~ /^\d+$/
       when :boolean
-        return true if text =~ /(true|t|yes|y|1)$/i
-        false if text =~ /(false|f|no|n|0)$/i
+        return true if text =~ /^(true|t|yes|y|1)$/i
+        false if text =~ /^(false|f|no|n|0)$/i
```

Comment:
I agree with @jhass here, it's better codestyle and I also used `\A`/`\z` at other places, but not everywhere (so it's my fault), so we should do it everywhere unless we need the line behavior. :) Thanks for your PR.